extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/CamelNativeImageProcessor.java

```diff
@@ -139,6 +139,11 @@ void reflectiveItems(
                 .map(ai -> ai.target().asMethod())
                 .forEach(mi -> reflectiveMethod.produce(new ReflectiveMethodBuildItem(mi)));
 
+        reflectiveClass.produce(
+                ReflectiveClassBuildItem.builder(
+                        "org.apache.camel.support.AbstractExchange",
+                        org.apache.camel.support.MessageSupport.class.getName())
```

Comment:
Have you checked whether there are possibly other classes that should be registered in order for the simple language to work as expected on the `exchange`?

Reply:
No, I have not. Do you have any other scenarios in mind this is not covering?

Reply:
None I found. As I wrote in #4860 :

> ... This, however, begs the question: what else should be registered? We should analyze all relevant classes used by the `simple(...)` language and ensure that they all work as expected.

Reply:
Thanks for explaining. Yeah, it would be nice to audit whether we need more. While I do not feel like I have free capacity do that these days, we can still keep the issue open.